connector_alndata/models/crm_lead.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (C) 2019 Open Source Integrators
+# <https://www.opensourceintegrators.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+import requests
+
+from odoo import api, fields, models, _
+from odoo.exceptions import RedirectWarning
+
+_logger = logging.getLogger(__name__)
+
+
+class Lead(models.Model):
+    _inherit = "crm.lead"
+
+    lastdate_changed = fields.Datetime('LastDateChanged')
+    rowversion = fields.Char('Row Version')
+    industry_id = fields.Many2one('res.partner.industry', 'Market')
+
+    def aln_auth_login(self, data_key=''):
+        """Aln Connector.
+
+        This method is used to connect with ALN and fetch the data.
+        """
+        url = self.env['ir.config_parameter'].get_param('alndata.api.url')
+        api_key = self.env['ir.config_parameter'].get_param('alndata.api.key')
+
+        url = url + data_key
+        if api_key == '0':
+            action = self.env.ref('base.ir_config_list_action')
+            msg = _('Cannot find a ALN Data URL and API key, '
+                    'You should configure it. '
+                    '\nPlease go to System Parameters.')
+            raise RedirectWarning(msg, action.id,
+                                  _('Go to the configuration panel'))
+        params = (
+            ('apikey', api_key),
+            ('Accept', 'application/json')
+        )
+        if data_key == 'ManagementCompanies':
+            params = (
+                ('apikey', api_key),
+                ('Accept', 'application/json'),
+                ('$expand', 'Addresses,PhoneNumbers'),
+            )
+        try:
+            response = requests.get(url, params=params)
+        except Exception as e:
+            raise Warning(_(("%s.") % str(e)))
+        response.raise_for_status()
+        content = response.content.decode('utf8')
+        return content
```

Comment:
@nikitavaghela We should be using RowVersion somewhere, otherwise we'll get all the records all the time.

Reply:
yes I am working on this.